File: android/app/src/main/java/com/smartexpense/android/domain/usecase/auth/RegisterUseCase.java
package com.smartexpense.android.domain.usecase.auth;

import com.smartexpense.android.data.remote.dto.request.RegisterRequestDto;
import com.smartexpense.android.data.remote.dto.response.AuthResponseDto;
import com.smartexpense.android.domain.repository.AuthRepository;
import com.smartexpense.android.domain.repository.ResultCallback;

public class RegisterUseCase {
    private final AuthRepository repository;

    public RegisterUseCase(AuthRepository repository) {
        this.repository = repository;
    }

    public void execute(String email, String password, String displayName, ResultCallback<AuthResponseDto> callback) {
        RegisterRequestDto request = new RegisterRequestDto(email, password, displayName);
        repository.register(request, callback);
    }
}

File: android/app/src/main/kotlin/com/smartexpense/android/domain/usecase/auth/RegisterUseCase.kt
package com.smartexpense.android.domain.usecase.auth

import com.smartexpense.android.data.remote.dto.request.RegisterRequestDto
import com.smartexpense.android.data.remote.dto.response.AuthResponseDto
import com.smartexpense.android.domain.repository.AuthRepository

class RegisterUseCase(private val repository: AuthRepository) {
    suspend fun execute(name: String, email: String, password: String): Result<AuthResponseDto> {
        val request = RegisterRequestDto(name, email, password)
        return repository.register(request)
    }
}
